src/test/java/com/arkvis/orchid/OvulationPredictionTest.java

```diff
@@ -3,25 +3,39 @@
 import org.junit.jupiter.api.Test;
 
 import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNull;
 
 class OvulationPredictionTest {
 
     @Test
-    void should_returnNullDate_when_gettingNextOvulationDateOnEmptyCalendar() {
-        LocalDate nextOvulation = new PeriodCalendar(new PeriodPredictor(), new OvulationPredictor()).getNextOvulationDate();
-        assertNull(nextOvulation);
+    void should_returnCorrectWindow_when_gettingNextFertilityWindowOnCalendarThatHasPeriod() {
+        PeriodCalendar calendar = new PeriodCalendar(new PeriodPredictor(), new OvulationPredictor());
+        calendar.addPeriod(LocalDate.now());
+
+        LocalDate nextPeriodDate = calendar.getNextPeriodWindow().getStartDate();
+        LocalDate nextOvulationDate = nextPeriodDate.minusDays(OvulationPredictor.DEFAULT_DAYS_BEFORE_PERIOD);
+        LocalDate ovulationStartDay = nextOvulationDate.minusDays(5);
+        List<LocalDate> ovulationWindow = new ArrayList<>();
+
+        IntStream.rangeClosed(0, OvulationPredictor.DEFAULT_DAYS_BEFORE_OVULATION + 1)
+                .forEach(i -> ovulationWindow.add(ovulationStartDay.plusDays(i)));
+
+        FertilityWindow nextFertilityWindow = calendar.getNextFertilityWindow();
+
+        assertEquals(ovulationWindow, nextFertilityWindow.getFertilityWindow());
+        assertEquals(nextOvulationDate, nextFertilityWindow.getOvulationDate());
     }
 
     @Test
-    void should_returnCorrectDate_when_gettingNextOvulationDateOnCalendarThatHasPeriod() {
+    void should_returnNullDate_when_gettingNextFertilityWindowOnEmptyCalendar() {
         PeriodCalendar calendar = new PeriodCalendar(new PeriodPredictor(), new OvulationPredictor());
-        calendar.addPeriod(LocalDate.now());
+        FertilityWindow fertilityWindow = calendar.getNextFertilityWindow();
 
-        LocalDate nextPeriodDate = calendar.getNextPeriodWindow().getStartDate();
-        LocalDate nextOvulationDate = calendar.getNextOvulationDate();
-        assertEquals(nextPeriodDate.minusDays(OvulationPredictor.DEFAULT_DAYS_BEFORE_PERIOD), nextOvulationDate);
+        assertNull(fertilityWindow);
```

Comment:
Instead of returning a null FertilityWindow let's return an empty FertilityWindow. That way the user of this service won't need to do null checks. 

You can have a method in FertilityWindow called isEmpty which returns true if it has no dates.